core/iwasm/interpreter/wasm_loader.c

```diff
@@ -3095,13 +3133,15 @@ load_import_section(const uint8 *buf, const uint8 *buf_end, WASMModule *module,
                         read_leb_uint32(p, p_end, u32);
                     module->import_table_count++;
 
-#if WASM_ENABLE_REF_TYPES == 0 && WASM_ENABLE_GC == 0
                     if (module->import_table_count > 1) {
+#if WASM_ENABLE_REF_TYPES == 0 && WASM_ENABLE_GC == 0
                         set_error_buf(error_buf, error_buf_size,
                                       "multiple tables");
                         return false;
-                    }
+#elif WASM_ENABLE_WAMR_COMPILER != 0
+                        module->is_ref_types_used = true;
```

Comment:
Is it reasonable to indicate ref type usage according to import table count?

Reply:
I think it should be reasonable, multiple tables are only allowed when reference types is enabled, which introduces table.init/table.fill/table.copy/table.get/table.set opcodes to operate on the specify table(s). As you can see above, when REF_TYPES and GC are disabled, "multiple tables" error is reported.
https://github.com/WebAssembly/reference-types/blob/master/proposals/reference-types/Overview.md#language-extensions
![image](https://github.com/bytecodealliance/wasm-micro-runtime/assets/20393758/bd13c8b5-24ff-4c4c-967f-29f331d60552)

Reply:
Got it, thanks